### PR TITLE
Create function that return the instalments array using country as pa…

### PR DESCRIPTION
**Pull request by Ruanito**
…rameters

### CODE REVIEW

File: src/Services/Gateways/BaseGateway.php [hunk 1]
```diff
@@ -10,11 +10,30 @@
 
 abstract class BaseGateway extends HttpService
 {
+    const COUNTRY_BRAZIL   = 'Brazil';
```
Comment (SparK-Cruz) [inline src/Services/Gateways/BaseGateway.php]: There's a model called Country which has the constants already defined there on [line 7](https://github.com/ebanx/benjamin/blob/master/src/Models/Country.php#L7).

File: src/Services/Gateways/BaseGateway.php [hunk 2]
```diff
@@ -10,11 +10,30 @@
...
+    const COUNTRY_COLOMBIA = 'Colombia';
+    const COUNTRY_MEXICO   = 'Mexico';
+    const COUNTRY_ARGENTINA = 'Argentina';
+
     /**
      * @var Exchange
      */
     protected $exchange;
 
+    public static $maxInstalmentCountry = [
```
Comment (SparK-Cruz) [inline src/Services/Gateways/BaseGateway.php]: It's advised to write a method on [CreditCard class](https://github.com/ebanx/benjamin/blob/master/src/Services/Gateways/CreditCard.php) returning a constant which then can be overridden on its specializations for each country.
Instalments don't belong in the BaseGateway as not every gateway has instalments.

File: src/Services/Gateways/BaseGateway.php [hunk 3]
```diff
@@ -10,11 +10,30 @@
...
+        self::COUNTRY_BRAZIL   => 12,
+        self::COUNTRY_COLOMBIA => 36,
+        self::COUNTRY_MEXICO   => 12,
+        self::COUNTRY_ARGENTINA => 12,
+    ];
+
+    public static $instalmentIncrementCountry = [
```
Comment (SparK-Cruz) [inline src/Services/Gateways/BaseGateway.php]: Same as above, make this a method in CreditCard class, not here.

File: src/Services/Gateways/BaseGateway.php [hunk 4]
```diff
@@ -83,4 +102,27 @@ protected function availableForCountryOrThrow($country)
                 : ' using '.$this->config->baseCurrency
         ));
     }
+
+    /**
+     * @param $country
+     * @return array
+     */
+    public static function getInstalmentsByCountry($country)
```
Comment (SparK-Cruz) [inline src/Services/Gateways/BaseGateway.php]: With the above changes this method goes to waste.

File: tests/Unit/Services/Gateways/BaseGatewayTest.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Tests\Unit\Services\Gateways;
+
+use Ebanx\Benjamin\Services\Gateways\BaseGateway;
+
+class BaseGatewayTest extends GatewayTestCase
+{
+    public function testInstalmentsArgentina()
```
Comment (SparK-Cruz) [inline tests/Unit/Services/Gateways/BaseGatewayTest.php]: Move these to [CreditCardTest](https://github.com/ebanx/benjamin/blob/master/tests/Unit/Services/Gateways/CreditCardTest.php), instalments really don't belong here.

File: tests/Unit/Services/Gateways/CreditCardTest.php [hunk 6]
```diff
@@ -255,6 +255,84 @@ public function testPaymentTermsMerchantTaxFlagOff()
...
+        $paymentTerms = $gateway->getPaymentTermsForCountryAndValue($country, $value);
+        $this->assertEquals(
+            1,
+            count($paymentTerms)
+        );
+    }
+
+    public function testInstalmentsArgentina()
+    {
+        $country = 'Argentina';
```
Comment (frop) [inline tests/Unit/Services/Gateways/CreditCardTest.php]: Please, make use of `Country`constants.

**Pull request merged**